Modules/Microsoft365DSC/DSCResources/MSFT_IntuneMobileAppsMacOSLobApp/MSFT_IntuneMobileAppsMacOSLobApp.schema.mof

```diff
@@ -0,0 +1,32 @@
+[ClassVersion("1.0.0"), FriendlyName("IntuneMobileAppsMacOSLobApp")]
+class MSFT_IntuneMobileAppsMacOSLobApp : OMI_BaseResource
+{
+    [Key, Description("The unique identifier for an entity. Read-only. Inherited from mobileApp object.")] String Id;
+
+    [Write, Description("Optional Response Headers Variable.")] String ResponseHeadersVariable;
+    [Write, Description("Additional Parameters.")] Hashtable AdditionalProperties; #TODOK
+
+    [Write, Description("The description of the app. Inherited from mobileApp.")] String Description;
+    [Write, Description("Name of the device health script publisher")] String Developer;
+    [Write, Description("The admin provided or imported title of the app. Inherited from mobileApp.")] String DisplayName;
+    [Write, Description("Name of the device health script publisher")] String InformationUrl;
+    [Write, Description("The value indicating whether the app is marked as featured by the admin. Inherited from mobileApp.")] String IsFeatured;
+    [Write, Description("The large icon to be displayed in the app details and used for upload of the icon. Inherited from mobileApp.")] String LargeIcon; #TODOK
```

Comment:
Can you please point me to a good example of code that shows how to deal with similar MIME content?

Reply:
We don't yet have something that similar, but something like the following would do it: 

```powershell
$mobileApp = Get-MgBetaDeviceAppManagementMobileApp -MobileAppId <ID>
$base64 = [System.Convert]::ToBase64String($mobileApp.LargeIcon.Value)
```

Now, I think that the `mimeContent` must be a separate CIM object specified in the MOF template. Then, in the get and export cmdlets, we need to convert that to the corresponding type. For that, you can take a look at how the assignments are implemented, these are always a separate CIM type that needs to be converted.

Reply:
So I exported the LargeIcon in the config file, and the M365Config.ps1 file compiled successfully as well. However the 
Start-dscconfiguration cmdlet is throwing an error due to very big size of the object. Do you know any work around? If not, then I guess, I will have to remove LargeIcon from this DSC, maybe that's why no other files contain similar example?


PS E:\3\a> Start-dscconfiguration  M365TenantConfig -Wait -verbose -force
VERBOSE: Perform operation 'Invoke CimMethod' with following parameters, ''methodName' = SendConfigurationApply,'className' = MSFT_DSCLocalConfigurationManager,'namespaceName' =
root/Microsoft/Windows/DesiredStateConfiguration'.
The WinRM client sent a request to the remote WS-Management service and was notified that the request size exceeded the configured MaxEnvelopeSize quota.
    + CategoryInfo          : LimitsExceeded: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : HRESULT 0x80338111
    + PSComputerName        : localhost

VERBOSE: Operation 'Invoke CimMethod' complete.
VERBOSE: Time taken for configuration job to complete is 1.068 seconds

Reply:
You have to increase the `maxEnvelopeSize` for WinRM to allow more content to be sent to it: 
```powershell
New-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WSMAN\Client' -Name maxEnvelopeSize -Value 65536 -PropertyType DWORD -Force
```

Reply:
ok, great, will try it, thank you.

Reply:
@Kajalp1079 I received an email notification about an issue with M365TenantConfig.ps1 not running because it can't find the `IntuneMobileAppsMacOSLobApp` resource. For the file to compile, you have to copy the directory `MSFT_IntuneMobileAppsMacOSLobApp` to the Microsoft365DSC DSCResources directory under `C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.24.1002.1\DSCResources`. If you are using another version, switch the version number accordingly. After that, it will compile.